### ISSUE TITLE
[nw13][win10] App window won't display - manifest name data directory 'poisoned'

### ISSUE THREAD

**Issue by tshinnic** [nwjs/nw.js#3829]
TL;DR : An old NW.js "data directory" can contain trash that will prevent your app from running.

This is going to sound crazy, but it is the only variable I can find after days of mimsy hunting.  The same app would not run (no window) using nwjs 0.13.0-alpha3 under Win10, but would run fine using all other combinations of 0.12.3 and 0.13.0-alpha3 and Win10, Win7, and Ubuntu.

I was working on an app using 0.12.3 and then decided to see what happened under 0.13.0-alpha3.  No window at all and see two processes in task manager on Win10.  Retried running the app under 0.12.3 on Win10 and all's well.  Retry under Win7 and then Ubuntu with both 0.12.3 _and_ 0.13.0-alpha3 and app window appears and gets same good results.  Only the combination of Win10 and 0.13.0-alpha3 was poisoned.

Accidently noticed a nwjs manual test app working fine with Win10/0.13.0a3.  And so began to compare searching for the critical differences in my html and js.  No problems found. Began to play with what parameters were specified in the package.manifest json file, thinking maybe something included like `"single-instance":  false,` was a problem.

Nothing significant until I played with the `"name"` parameter in package.json!  If I changed 

```
            "name": "test01",
```

to

```
            "name": "test0z",
```

everything started working!  Worked backwards until all sources were identical to the starting points except for this one difference which made the app work.

I even suspected anti-virus scanning and ran tests with those turned off, and that didn't help.

Wondering why this `"name"` was poison I tried looking where else 'test01' was mentioned in the sources, and played around with changing those.  No changes to behavior _until_ I noticed the directory name I was developing within:

```
        C:\Toms\Study\Javascript\NWjs\test01
```

Changed that last name to `test0z` and **poof** everything started working! Again with `...\test01` and `"test01"` and failure.  Tried with `...\test0z` and `"test0z"` and **_success**_!!?!? (cue weeping with frustration)  Several other variations of directory name and manifest `"name"` worked fine, but `...\test01` and `"test01"` failed always.

Then tried another variation - `"name": "test01 ",`. With the added space I got an interesting error message:

```
        NW.js cannot read and write to its data directory:
        C:\Users\Tom\AppData\Local\test01 \User Data
```

which pointed me to another place the name was being used.

Changed `C:\Users\Tom\AppData\Local\test01\` to `C:\Users\Tom\AppData\Local\test01.old\` and reran the app, which now succeeded with `...\test01\` and `"test01"`.  And now there was a _newer_ `C:\Users\Tom\AppData\Local\test01\` created.

Something was poisoned in the nwjs "data directory" whose name was based on the manifest `"name"`!  That is what made the name 'test01' so toxic - leftover trash from some previous execution!  No other symptoms than the app window not appearing and only two processes instead of the normal three processes.

Once I'd renamed the "data directory" the old bad contents were no longer preventing the app from running.  Who knew that a data directory existed, or could be poisoned?

Note: I still have the poisoned data directory if someone wants me to check some kind of logs for debug messages...
##

**Comment by maxbarry** [comment]
Not 100% sure it's the same issue, but on Linux when I first tried running my 0.12.3 app in 0.13.0, I had the same experience: no app, no window, no useful logs, and it only started working after I deleted the data directory ~/.config/<appname>/

**Comment by Christywl** [comment]
Is this issue still reproduced on the latest nwjs build?

**Comment by Christywl** [comment]
close until there is more information.
